aws/resource_aws_cognito_user_pool_domain_test.go

```diff
@@ -13,6 +14,65 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_cognito_user_pool_domain", &resource.Sweeper{
+		Name: "aws_cognito_user_pool_domain",
+		F:    testSweepCognitoUserPoolDomains,
+	})
+}
+
+func testSweepCognitoUserPoolDomains(region string) error {
```

Comment:
i'm not hp sure if this functionality should be implemented here on it's own or added into the `user_pool` sweeper function that already exists as they overlap quite a bit

Reply:
We've tended to lean in the direction that this is implemented (splitting out the sweepers), I think mostly to:

* Keep the sweeper deletion logic near its resource logic
* Allow us in the future to disable certain bits more easily
* Allow us in the future to acceptance test just this sweeper functionality

It does certainly feel a little weird that you list through the parent resource in this case. 😅